src/main/java/codechicken/nei/ItemPanel.java

```diff
@@ -1,85 +1,83 @@
 package codechicken.nei;
 
-import codechicken.lib.vec.Rectangle4i;
 import codechicken.nei.api.GuiInfo;
 import codechicken.nei.api.INEIGuiHandler;
 import codechicken.nei.guihook.GuiContainerManager;
+import codechicken.nei.ItemsGrid.ItemPanelSlot;
 import codechicken.nei.recipe.GuiCraftingRecipe;
 import codechicken.nei.recipe.GuiRecipe;
 import codechicken.nei.recipe.GuiUsageRecipe;
+import codechicken.nei.recipe.StackInfo;
 import net.minecraft.client.gui.inventory.GuiContainer;
+import net.minecraft.client.gui.inventory.GuiContainerCreative;
 import net.minecraft.inventory.Slot;
 import net.minecraft.item.ItemStack;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.IFluidContainerItem;
 
 import java.util.ArrayList;
 
-import static codechicken.lib.gui.GuiDraw.drawRect;
-import static codechicken.nei.NEIClientConfig.canPerformAction;
-
-public class ItemPanel extends Widget {
-    /**
-     *  Backwards compat :-/
-     */
-    public static ArrayList<ItemStack> items = new ArrayList<>();
-
-    /**
-     * Should not be externally modified, use updateItemList
-     */
-    public ArrayList<ItemStack> realItems = new ArrayList<>();
-    /**
-     * Swapped into visible items on update
-     */
-    protected ArrayList<ItemStack> _items = realItems;
-
-    public ArrayList<ItemStack> getItems() {
-        return realItems;
-    }
+public class ItemPanel extends Widget
+{
 
-    public static void updateItemList(ArrayList<ItemStack> newItems) {
-        ItemPanels.itemPanel._items = newItems;
-    }
+    public ItemStack draggedStack;
+    public int mouseDownSlot = -1;
+
+    public Button pagePrev;
+    public Button pageNext;
+    public Label pageLabel;
 
-    public class ItemPanelSlot
```

Comment:
we can't remove public classes, they're likely used by other addons.

Reply:
You can neuter them, or redirect them to the new replacement, but we need to maintain backwards compat with other addons.

Reply:
I do not know how to do that. can you help me?

Reply:
This class move in `Items Grid` because it create instances with this class